Lib/warnings.py

```diff
@@ -701,17 +707,48 @@ def extract():
 # If either if the compiled regexs are None, match anything.
 try:
     from _warnings import (filters, _defaultaction, _onceregistry,
-                           warn, warn_explicit, _filters_mutated)
+                           warn, warn_explicit,
+                           _filters_mutated_unlocked,
+                           _acquire_lock, _release_lock,
+    )
     defaultaction = _defaultaction
     onceregistry = _onceregistry
     _warnings_defaults = True
+
+    class _Lock:
+        def __enter__(self):
+            _acquire_lock()
+            return self
+
+        def __exit__(self, *args):
+            _release_lock()
+
+    _lock = _Lock()
+
+    def _filters_mutated():
+        # Even though this function is part of the public API, it's used
+        # by a fair amount of user code.
+        with _lock:
+            _filters_mutated_unlocked()
```

Comment:
I suggest to name it as  `_filters_mutated_lock_held` instead, that's the name used often in free-threading. 
`_filters_mutated_unlocked` seems to suggest that lock should be in unlocked/released state which isn't correct.

Reply:
There is other code that uses the _unlocked suffix for the same purpose but I like your suggested name better. I'll change it. Saying "unlocked" could definitely be confusing.